### ISSUE TITLE
valid-jsdoc misses issues in function expression docs

### ISSUE THREAD

**Issue by tschaub** [eslint/eslint#4964]
With ESLint @ 86d64ca571faa7260233d41b8eb0391a51d59ba0, it looks like the `valid-jsdoc` rule misses issues in documentation for function expressions.

The `script.js` below provides two invalid test cases (both the function declaration and the function expression have a typo in the `@param` name).

```js
/* eslint valid-jsdoc: [2, {requireReturn: false}] */

// valid-jsdoc catches the typo below

/**
 * Check the param name.
 * @param {string} typoInTheName The param description.
 * @return {string} The return description.
 */
function foo(bar) {
  return bar;
}

// valid-jsdoc does not catch the typo below

check(
  /**
   * Check the param name.
   * @param {string} typoInTheName The param description.
   * @return {string} The return description.
   */
  function(bar) {
    return bar;
  });
```

Here's what I get running this through ESLint:
```
$ node bin/eslint.js --no-eslintrc script.js 

/Users/tschaub/projects/eslint/script.js
  5:1  error  Expected JSDoc for 'bar' but found 'typoInTheName'  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)
```

Instead of 1 problem, I would expect two.  The `valid-jsdoc` rule correctly catches the typo in the `@param` name for the function declaration, but misses the same for the function expression.

**Comment by tschaub** [comment]
It looks like this is specific to a function expression in a call (or new) expression.  See https://github.com/eslint/eslint/compare/master...tschaub:issue-4964 for a proposed fix.  I'll add tests if others agree this looks like the right direction.

**Comment by nzakas** [comment]
@gyandeeps ?

**Comment by gyandeeps** [comment]
Sounds good to me. Not very common case I guess but I am fine with this recommendation.
